mission-7/schema.sql

```diff
@@ -0,0 +1,160 @@
+BEGIN;
+
+CREATE TABLE "public"."Product"(
+    id SERIAL PRIMARY KEY,
+    name TEXT NOT NULL,
+    description TEXT NOT NULL,
+    price INTEGER NOT NULL,
+    tags TEXT[] NOT NULL,
+    createdAt TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    updatedAt TIMESTAMP(3) NOT NULL,
+
+    userId INTEGER
+);
+
+CREATE TABLE "public"."Article"(
+    id SERIAL PRIMARY KEY,
+    title TEXT NOT NULL,
+    content TEXT NOT NULL,
+    createdAt TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    updatedAt TIMESTAMP(3) NOT NULL,
+
+    userId INTEGER
+);
+
+CREATE TABLE "public"."Comment"(
+    id SERIAL PRIMARY KEY,
+    content TEXT NOT NULL,
+    createdAt TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    updatedAt TIMESTAMP(3) NOT NULL,
+
+    userId INTEGER,
+    productId INTEGER,
+    articleId INTEGER
+);
+
+CREATE TABLE "public"."Image"(
+    id SERIAL PRIMARY KEY,
+    publicId TEXT UNIQUE NOT NULL,
+    url TEXT UNIQUE NOT NULL,
+    createdAt TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    updatedAt TIMESTAMP(3) NOT NULL,
+    
+    userId INTEGER UNIQUE,
+    productId INTEGER,
+    articleId INTEGER
+);
+
+CREATE TABLE "public"."User"(
+    id SERIAL PRIMARY KEY,
+    email VARCHAR(255) UNIQUE NOT NULL,
+    nickname VARCHAR(50) UNIQUE NOT NULL,
+    password VARCHAR(255) NOT NULL,
+    refreshToken VARCHAR(512) UNIQUE, 
+    createdAt TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    updatedAt TIMESTAMP(3) NOT NULL
+);
```

Comment:
간편 로그인에(소셜 로그인) 대한 설계가 빠져있는 것 같아요!!
보통 소셜(provider) 플랫폼의 타입과 OAUTH에서 제공해주는 id를 활용합니다.

예를들면

```suggestion
CREATE TABLE "public"."User"(
    id SERIAL PRIMARY KEY,
    email VARCHAR(255) UNIQUE NOT NULL,
    nickname VARCHAR(50) UNIQUE NOT NULL,
    password VARCHAR(255) NOT NULL,
    refreshToken VARCHAR(512) UNIQUE, 
    providerId VARCHAR(255),
    providerType VARCHAR(255),
    createdAt TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
    updatedAt TIMESTAMP(3) NOT NULL,

    -- 소셜 로그인 고유성 보장
    CONSTRAINT user_provider_unique UNIQUE (providerType, providerId),

    -- providerType / providerId 둘 다 NULL 또는 둘 다 NOT NULL만 허용
    CONSTRAINT user_provider_check CHECK (
        (providerType IS NULL AND providerId IS NULL)
        OR (providerType IS NOT NULL AND providerId IS NOT NULL)
    )
);
```

혹은 별도의 테이블로 관리할 수 있어요.